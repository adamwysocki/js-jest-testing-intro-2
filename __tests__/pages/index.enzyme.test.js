import React from 'react';
import { shallow, mount } from 'enzyme';

import Index from '../../pages/index';

describe("Enzyme tests", () => {
    it('Should render correctly with no props', () => {
        const component = shallow(<Index />);
        expect(component).toMatchSnapshot();
    });

    it('Should change to spanish when language is toggled to true', () => {
        const component = mount(<Index />);
        component
            .find('input[type="checkbox"]')
            .simulate('change', {currentTarget: {checked: true}});

        expect(component).toMatchSnapshot();
        component.unmount();
    });
});