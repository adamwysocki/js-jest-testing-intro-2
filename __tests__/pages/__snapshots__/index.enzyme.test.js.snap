// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enzyme tests Should change to spanish when language checkbox is toggled to true 1`] = `
<Index>
  <div
    className="container"
    style={
      Object {
        "marginTop": "150px",
      }
    }
  >
    <SideEffect(Head)>
      <Head />
    </SideEffect(Head)>
    <h1>
      ¡Hola!
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <input
          aria-describedby="nameHelp"
          className="form-control"
          id="name"
          onChange={[Function]}
          placeholder="Ingrese su nombre ..."
          type="text"
          value=""
        />
      </div>
      <div
        className="form-check"
      >
        <input
          checked={true}
          className="form-check-input"
          id="espanol"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="espanol"
        >
          español
        </label>
      </div>
    </form>
  </div>
</Index>
`;

exports[`Enzyme tests Should render correctly with no props 1`] = `
<div
  className="container"
  style={
    Object {
      "marginTop": "150px",
    }
  }
>
  <SideEffect(Head)>
    <title>
      React Testing w/ Jest Part 2
    </title>
    <link
      href="https://bootswatch.com/4/cerulean/bootstrap.css"
      rel="stylesheet"
    />
  </SideEffect(Head)>
  <h1>
    Hello!
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        aria-describedby="nameHelp"
        className="form-control"
        id="name"
        onChange={[Function]}
        placeholder="Enter name ..."
        type="text"
        value=""
      />
    </div>
    <div
      className="form-check"
    >
      <input
        checked={false}
        className="form-check-input"
        id="espanol"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="espanol"
      >
        español
      </label>
    </div>
  </form>
</div>
`;

exports[`Enzyme tests Should update the welcome header when a name is typed 1`] = `
<Index>
  <div
    className="container"
    style={
      Object {
        "marginTop": "150px",
      }
    }
  >
    <SideEffect(Head)>
      <Head />
    </SideEffect(Head)>
    <h1>
      Hello Adam Wysocki!
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <input
          aria-describedby="nameHelp"
          className="form-control"
          id="name"
          onChange={[Function]}
          placeholder="Enter name ..."
          type="text"
          value="adam wysocki"
        />
      </div>
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-input"
          id="espanol"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="espanol"
        >
          español
        </label>
      </div>
    </form>
  </div>
</Index>
`;

exports[`Enzyme tests Should update the welcome header when a name is typed and language checkbox is toggled to true 1`] = `
<Index>
  <div
    className="container"
    style={
      Object {
        "marginTop": "150px",
      }
    }
  >
    <SideEffect(Head)>
      <Head />
    </SideEffect(Head)>
    <h1>
      ¡Hola Adam Wysocki!
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <input
          aria-describedby="nameHelp"
          className="form-control"
          id="name"
          onChange={[Function]}
          placeholder="Ingrese su nombre ..."
          type="text"
          value="adam wysocki"
        />
      </div>
      <div
        className="form-check"
      >
        <input
          checked={true}
          className="form-check-input"
          id="espanol"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="espanol"
        >
          español
        </label>
      </div>
    </form>
  </div>
</Index>
`;
